### PR TITLE
Add event detail pages

### PR DESCRIPTION
**Pull request by briaguya**
https://github.com/cityofaustin/techstack/issues/3816
https://github.com/cityofaustin/techstack/issues/3815

# Description

tl;dr - Add event detail pages

* Updates janis to build event detail pages: https://github.com/cityofaustin/techstack/issues/3816
* Makes event detail pages look like the designs/act according to  in: https://github.com/cityofaustin/techstack/issues/3815

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 
     
https://janis-3816-build-event-pages.netlify.com/en/event/2020/2/6/valentines-day-dance/

more example events with working view live/preview links here: https://joplin-pr-3814-events-non-joyo.herokuapp.com/admin/

# Checklist:
- [x] Request reviewers
- [x] Slack-out message for review
- [x] Link this PR in the issue
- [x] Moved card to *review* in zenhub

### CODE REVIEW

File: src/components/Contact/_Contact.scss [hunk 1]
```diff
@@ -165,7 +165,7 @@ This will enable proper line wrapping for the :last-child.
   }
 }
 
-.coa-ContactDetails__location-link {
+@mixin location-link($text-class, $arrow-class, $arrow-size) {
```
Comment (briaguya) [inline src/components/Contact/_Contact.scss]: made this a mixin so we can use it for EventDetails too

File: src/components/HtmlFromRichText/_HtmlFromRichText.scss [hunk 2]
```diff
@@ -1,8 +1,8 @@
-.coa-HtmlFromAdmin {
+.coa-HtmlFromRichText {
```
Comment (briaguya) [inline src/components/HtmlFromRichText/_HtmlFromRichText.scss]: finally getting around to doing this

File: src/components/Pages/Event/EventDetailCard/index.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      <div className="coa-EventDetailItem">
+        <i className="material-icons">event</i>
+        <div className="coa-EventDetailItem__content">
+          <div className="coa-EventDetailItem__date">
+            {moment(date, 'YYYY-MM-DD').format('dddd • LL')}
+          </div>
+          <EventTime startTime={startTime} endTime={endTime} />
+        </div>
+      </div>
+      {location && location.locationType === 'city_location' ? (
```
Comment (briaguya) [inline src/components/Pages/Event/EventDetailCard/index.js]: we could probably do something with the API on joplin to only return one or the other here, so instead we had:
```
location {
  locationType
  title
  city
  street
}
```

but for now:
```
location {
  locationType
  cityLocation {
    stuff
  }
  remoteLocation {
     stuff
  }
  additionalDetails
}
```
works

File: src/js/i18n/locales/es.json [hunk 4]
```diff
@@ -125,5 +125,12 @@
...
   "userFeedback.characterLimit": "Límite de caracteres",
   "userFeedback.of": "de",
-  "userFeedback.errorMessage": "Lo sentimos mucho, tuvimos problemas para recibir sus comentarios. Por favor inténtelo de nuevo o envíe un correo electrónico a"
+  "userFeedback.errorMessage": "Lo sentimos mucho, tuvimos problemas para recibir sus comentarios. Por favor inténtelo de nuevo o envíe un correo electrónico a",
+  "events.canceled": "Cancelado",
+  "events.free": "Gratis",
+  "events.register": "Regístrese",
+  "events.public": "Abierto al público",
+  "events.details": "Detalles",
+  "events.allEvents": "Todos eventos",
```
Comment (chiaberry) [inline src/js/i18n/locales/es.json]: i will check if this is right, i want to say its "Todos los eventos"

File: src/components/Pages/Event/EventDetailCard/index.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        <EventLocationDetail
+          name={location.remoteLocation.name}
+          street={location.remoteLocation.street}
+          city={location.remoteLocation.city}
+          state={location.remoteLocation.state}
+          zip={location.remoteLocation.zip}
+          unit={location.remoteLocation.unit}
+          additionalDetails={location.additionalDetails}
+        />
+      ) : null}
```
[the remark this answers is not in the document]
Comment (briaguya) [reply src/components/Pages/Event/EventDetailCard/index.js]: yeah, that's definitely a way we can do that, I'm pretty sure I just added the check to both when I ran into an error with an event missing a location.

the only thing missing from your example that comes to mind is an explicit check for `locationType == remote_location`

if we add that check back it we'd end up with:
```
location && 
  (location.locationType == city ?
     city code : 
    (location.locationType == remote ? 
      remote code :
      null)
  )
```
instead of the current
```
  location && location.locationType == city ?
     city code : 
    location && location.locationType == remote ? 
      remote code :
      null)
```

File: src/js/helpers/cleanData.js [hunk 6]
```diff
@@ -720,3 +720,19 @@ export const cleanFormContainersForPreview = allFormContainers => {
 
   return form;
 };
+
```
Comment (easherma) [inline src/js/helpers/cleanData.js]: explanation about what this is for would be great; it seems kinda tangental to the events page stuff but aimed at cleaning up some stuff for previews & IA stuff?

File: src/js/helpers/location.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+export const getEncodedLocation = location => {
```
Comment (easherma) [inline src/js/helpers/location.js]: this is purely for sticking address fields together?

File: src/components/Pages/Event/EventDetailCard/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (chiaberry) [reply src/components/Pages/Event/EventDetailCard/index.js]: you're right -- I was thinking we could avoid explicitly checking for remote_location, since there are only two options anyway. I rescind my refactor!

File: src/js/i18n/locales/es.json (hunk 4 above)
Comment (chiaberry) [reply src/js/i18n/locales/es.json]: just confirmed with my mom & sister, it is "todos los eventos" and not "todos eventos"

File: src/components/Pages/Event/EventDetailCard/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (briaguya) [reply src/components/Pages/Event/EventDetailCard/index.js]: alright, let's use reactions to vote on the actual code (jsprettier formatted):

### love it as is ❤️
```
      {location && location.locationType === 'city_location' ? (
        <EventLocationDetail
          name={location.cityLocation.title}
          street={location.cityLocation.physicalStreet}
          city={location.cityLocation.physicalCity}
          state={location.cityLocation.physicalState}
          zip={location.cityLocation.physicalZip}
          unit={location.cityLocation.physicalUnit}
          additionalDetails={location.additionalDetails}
        />
      ) : location && location.locationType === 'remote_location' ? (
        <EventLocationDetail
          name={location.remoteLocation.name}
          street={location.remoteLocation.street}
          city={location.remoteLocation.city}
          state={location.remoteLocation.state}
          zip={location.remoteLocation.zip}
          unit={location.remoteLocation.unit}
          additionalDetails={location.additionalDetails}
        />
      ) : null}
```

### refactor rocketship 🚀

```
      {!!location && location.locationType === 'city_location' ? (
        <EventLocationDetail
          name={location.cityLocation.title}
          street={location.cityLocation.physicalStreet}
          city={location.cityLocation.physicalCity}
          state={location.cityLocation.physicalState}
          zip={location.cityLocation.physicalZip}
          unit={location.cityLocation.physicalUnit}
          additionalDetails={location.additionalDetails}
        />
      ) : location.locationType === 'remote_location' ? (
        <EventLocationDetail
          name={location.remoteLocation.name}
          street={location.remoteLocation.street}
          city={location.remoteLocation.city}
          state={location.remoteLocation.state}
          zip={location.remoteLocation.zip}
          unit={location.remoteLocation.unit}
          additionalDetails={location.additionalDetails}
        />
      ) : null}
```

[the remark this answers is not in the document]
Comment (briaguya) [reply src/components/Pages/Event/EventDetailCard/index.js]: actually, I'm voting for a third option, just a sec

File: src/js/helpers/location.js (hunk 7 above)
Comment (briaguya) [reply src/js/helpers/location.js]: it's what we were already using for our staticmap links, so I moved it to a helper file

File: src/components/Pages/Event/EventDetailCard/index.js (hunk 3 above)
Comment (briaguya) [reply src/components/Pages/Event/EventDetailCard/index.js]: added a note to the fragment in queries

File: src/js/helpers/cleanData.js (hunk 6 above)
Comment (briaguya) [reply src/js/helpers/cleanData.js]: everything up there was just jsprettier formatting. The getOfferedByFromRelatedDepartments logic used to be hiding around a bunch of places in some form or another.

We use it in static.config.js, but previews don't use the logic in there, so I added it in.

We don't have the cleanest solutions around Janis for this stuff, and I don't have any good ideas for ways to clean it up at the moment
